src/NeoPixelAnimator.h

```diff
@@ -137,6 +138,12 @@ class NeoPixelAnimator
         _timeScale = (timeScale < 1) ? (1) : (timeScale > 32768) ? 32768 : timeScale;
     }
 
+    void setDuration(uint16_t indexAnimation, uint16_t newDuration);
```

Comment:
get/set methods are supposed to be lightweight property accessors.  This is not the case with your setDuration.  I believe this really should called be ChangeDuration
Further, you need to modify the keywords file to include the new public method.